presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGenerator.java

```diff
@@ -392,6 +395,68 @@ else if (fractionInput instanceof VariableReferenceExpression) {
+            }
+            else if (standardErrorInput instanceof VariableReferenceExpression) {
+                Selection fraction = inputSelections.get(standardErrorInput);
+                if (fraction.getOrigin() != LITERAL) {
+                    throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(),
+                            "Cannot handle approx_distinct standard error argument be a non literal " + aggregation);
+                }
+                standardErrorString = fraction.getDefinition();
+            }
+            else {
+                throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), "Expected the standard error to be a constant or a variable " + standardErrorInput);
+            }
+
+            double standardError;
+            try {
+                standardError = Double.parseDouble(standardErrorString);
+                if (standardError <= LOWEST_APPROX_DISTINCT_MAX_STANDARD_ERROR || standardError >= HIGHEST_APPROX_DISTINCT_MAX_STANDARD_ERROR) {
+                    throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), format("Cannot handle approx_distinct parsed as %d from input %s (function %s)",
+                            standardError, standardErrorString, aggregation));
+                }
+            }
+            catch (Exception e) {
+                throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), format("Cannot handle approx_distinct parsing to numerical value from input %s (function %s)",
+                        standardErrorString, aggregation));
+            }
+            // Pinot uses DISTINCTCOUNTHLL to do distinct count estimation, with hyperloglog algorithm.
+            //
+            // The HyperLogLog (HLL) data structure is a probabilistic data structure used to estimate the cardinality
+            // of a data set.
+            // In order to construct HLL data structure, the parameter log2m is used which represents the number of
+            // registers used internally by HLL.
+            //
+            // If we want a higher accuracy, we need to set these to higher values. Such a configuration
+            // will have additional overhead because our HLL will occupy more memory. If we're fine with lower accuracy,
+            // we can lower those parameters, and our HLL will occupy less memory.
+            //
+            // The relative standard deviation of HyperLoglog is:
+            //     rsd = 1.106 / sqrt(2^(log2m))
+            // So:
+            //     log2m = 2 * log(1.106 / rsd) / log(2)
+            int log2m = (int) (2 * Math.log(1.106 / standardError) / Math.log(2));
+            if (log2m < 1) {
+                throw new PinotException(PINOT_UNSUPPORTED_EXPRESSION, Optional.empty(), format("Cannot handle approx_distinct, the log2m generated from error is %d from input %s (function %s)",
```

Comment:
one param per line; leave the first line empty

Reply:
done.